src/components/dialogs/SpoolmanChangeSpoolDialog.vue

```diff
@@ -182,6 +189,26 @@ export default class SpoolmanChangeSpoolDialog extends Mixins(BaseMixin) {
 
     setSpool(spool: ServerSpoolmanStateSpool) {
         this.$store.dispatch('server/spoolman/setActiveSpool', spool.id)
+
+        // Close the dialog if no tool is selected
+        if (!this.tool) {
+            this.close()
+            return
+        }
+
+        // Set spool_id for tool
+        const gcode = `SET_GCODE_VARIABLE MACRO=${this.tool} VARIABLE=spool_id VALUE=${spool.id}`
+        this.$store.dispatch('server/addEvent', { message: gcode, type: 'command' })
+        this.$socket.emit('printer.gcode.script', { script: gcode })
+
+        // Close dialog if save_variables is not enabled
+        if (this.existsSaveVariables) this.close()
+
+        // Set spool_id to save_variable
+        const gcode2 = `SAVE_VARIABLE VARIABLE=${this.tool.toUpperCase()}__SPOOL_ID VALUE=${spool.id}`
```

Comment:
maybe we could check if the save_variable is available / configured. if not we could post a warning into the console.

Reply:
this check is already 3 lines above and will stop before this command.

Reply:
well I got  an unknown command error

Reply:
![grafik](https://github.com/user-attachments/assets/f730d7b3-c9c3-4b04-ab31-412f71182196)